pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -44,54 +93,79 @@ func (s *store) run(ctx context.Context) error {
 		// https://cloud.google.com/run/quotas#api
 		v, next, err := s.client.List(ctx, ops)
 		if err != nil {
-			return fmt.Errorf("failed to list cloudrun services: %w", err)
+			return nil, err
 		}
-		svc = append(svc, v...)
+		svcs = append(svcs, v...)
 		if next == "" {
 			break
 		}
 		cursor = next
 	}
-
-	// Update apps to the latest.
-	s.setApps(svc)
-
-	return nil
+	return svcs, nil
 }
 
-func (s *store) setApps(svc []*provider.Service) {
-	apps := make(map[string]provider.ServiceManifest, len(svc))
-	for i := range svc {
-		sm, err := svc[i].ServiceManifest()
-		if err != nil {
-			s.logger.Error("failed to load cloudrun service into service manifest", zap.Error(err))
+func (s *store) listRevisionsFromServices(ctx context.Context, svcs []*provider.Service) map[string][]*provider.Revision {
+	revs := make(map[string][]*provider.Revision, len(svcs))
+	for i := range svcs {
+		id, ok := svcs[i].UID()
+		if !ok {
 			continue
 		}
-		appID := sm.Labels()[provider.LabelApplication]
-		apps[appID] = sm
+		names := svcs[i].ActiveRevisionNames()
+		if rs := s.getMultiRevisions(ctx, names); len(rs) > 0 {
+			revs[id] = rs
+		}
 	}
-	s.apps.Store(apps)
+	return revs
+}
+
+func (s *store) getMultiRevisions(ctx context.Context, names []string) []*provider.Revision {
```

Comment:
nit: `fetchActiveRevisions` and return an error instead of logging inside it.
error handling should be done by its caller.